lizmap/modules/lizmap/classes/qgisExpressionUtils.class.php

```diff
@@ -0,0 +1,282 @@
+<?php
+/**
+ * @author    3liz
+ * @copyright 2020 3liz
+ *
+ * @see      http://3liz.com
+ *
+ * @license Mozilla Public License : http://www.mozilla.org/MPL/
+ */
+class qgisExpressionUtils
+{
+
+    /**
+     * Return criteria dependencies (fields) for a QGIS expression
+     *
+     * @param string $exp A QGIS expression string
+     *
+     * @return array The list of criteria dependencies
+     *
+     */
+    static public function getCriteriaFromExpression($exp)
+    {
+        if ($exp === null || $exp === '') {
+            return array();
+        }
+        preg_match_all('/"([^"]+)"/', $exp, $matches);
+        if (count($matches) < 2) {
+            return array();
+        }
+        return array_values(array_unique($matches[1]));
+    }
+
+    /**
+     * Return criteria dependencies (fields) for QGIS expressions
+     *
+     * @param array $expressions list of QGIS expressions
+     *
+     * @return array The list of criteria dependencies
+     *
+     */
+    static public function getCriteriaFromExpressions($expressions)
+    {
+        $criteriaFrom = array();
+        foreach ($expressions as $id => $exp) {
+            if ($exp === null || $exp === '') {
```

Comment:
Are we sure expression has been trimed before testing `$exp === ''` ? Not sure QGIS trims it automatically when user leaves a blank space or tab ?

Reply:
fixed